### PR TITLE
Update KernelAbstractions to use CUDA 1.0

### PR DESCRIPTION
**Pull request by jakebolewski**
Renamed CUDA <: GPU struct to CUDAGPU to avoid name conflicts with CUDA module.

### CODE REVIEW

**Comment by jakebolewski** [comment]
Part of https://github.com/CliMA/ClimateMachine.jl/issues/1228

File: docs/src/index.md [hunk 1]
```diff
@@ -55,11 +55,11 @@ all(A .== 2.0)
 1. Functions inside kernels are forcefully inlined, except when marked with `@noinline`.
 2. Floating-point multiplication, addition, subtraction are marked contractable.
 
-## Important differences to CUDAnative
+## Important differences to CUDA
```
Comment (vchuravy) [inline docs/src/index.md:58]:
```suggestion
## Important differences to CUDA.jl
```

File: src/KernelAbstractions.jl [hunk 2]
```diff
@@ -303,7 +303,7 @@ abstract type Device end
 abstract type GPU <: Device end
 
 struct CPU <: Device end
-struct CUDA <: GPU end
+struct CUDAGPU <: GPU end
```
Comment (vchuravy) [inline src/KernelAbstractions.jl:306]: I am not a fan of the name, maybe we rename it to `NVIDIA` like the prospective names below?

But I don't have a super strong opinion on this.

Comment (jakebolewski) [reply src/KernelAbstractions.jl:306]: yeah I have no issues changing it, I'll switch to the consensus.

NVIDIA wouldn't clash with CUDA and aligns with the other names so it seems like a good candidate.

Comment (jakebolewski) [reply src/KernelAbstractions.jl:306]: I was going to use `CUDABackend` but I'm not sure that is better as it's a little long

Comment (lcw) [reply src/KernelAbstractions.jl:306]: What names are you thinking about for the AMD and Intel GPU backends?

Comment (jakebolewski) [reply src/KernelAbstractions.jl:306]: it seems `AMD` and `Intel`?

Comment (jakebolewski) [reply src/KernelAbstractions.jl:306]: There are a lot of ` device isa CUDA` in the CLIMA code, so would `CUDADevice` be a better name?

Comment (lcw) [reply src/KernelAbstractions.jl:306]: Either way sounds okay with me.

**Comment by jakebolewski** [comment]
Renamed to `CUDADevice`.

**Comment by jakebolewski** [comment]
@vchuravy can you tag and register another version?

**Pull request merged**